lib/dep-graph-builders/npm-lock-v2/index.ts

```diff
@@ -235,85 +242,79 @@ const getChildNode = (
 
 export const getChildNodeKey = (
   name: string,
+  version: string,
   ancestry: { name: string; key: string; inBundle: boolean }[],
   pkgs: Record<string, NpmLockPkg>,
   pkgKeysByName: Map<string, string[]>,
 ): string | undefined => {
+  // This is a list of all our possible options for the childKey
   const candidateKeys = pkgKeysByName.get(name);
 
   // Lockfile missing entry
   if (!candidateKeys) {
     return undefined;
   }
 
-  // Only one candidate then just take it
+  // If we only have one candidate then we just take it
   if (candidateKeys.length === 1) {
     return candidateKeys[0];
   }
-
-  // If we are a bundled dep we have to choose candidate carefully
-  if (ancestry.length && ancestry[ancestry.length - 1].inBundle) {
-    const bundleRootIdx = ancestry.findIndex((el) => el.inBundle === true) - 1;
-    const ancestryFromBundleId = [
-      ...ancestry.slice(bundleRootIdx).map((el) => el.name),
-      name,
-    ];
-
-    const candidateAncestries = candidateKeys.map((el) =>
-      el.replace('node_modules/', '').split('/node_modules/'),
-    );
-
-    const filteredCandidates = candidateKeys.filter((candidate, idx) => {
-      return candidateAncestries[idx].every((pkg) => {
-        return ancestryFromBundleId.includes(pkg);
-      });
+  // If we are bundled we assume we are scoped by the bundle root at least
+  // otherwise the ancestry root is the root ignoring the true root
+  const isBundled = ancestry[ancestry.length - 1].inBundle;
+  const rootOperatingIdx = isBundled
+    ? ancestry.findIndex((el) => el.inBundle === true) - 1
+    : 1;
+  const ancestryFromRootOperatingIdx = [
+    ...ancestry.slice(rootOperatingIdx).map((el) => el.name),
+    name,
+  ];
+
+  // We filter on a number of cases
+  let filteredCandidates = candidateKeys.filter((candidate) => {
+    let keep = true;
```

Comment:
Would it make more sense to use ad-hoc variables that explains why you want to bail out instead of having one "keep" "global" variable?